base/pkg/entry.jl

```diff
@@ -582,8 +582,8 @@ function build!(pkgs::Vector, errs::Dict, seen::Set=Set())
             end
         end
     catch err
-        kill(pobj)
         close(io)
+        isa(err, PkgError) ? wait(pobj) : kill(pobj)
```

Comment:
I think since `PkgError` doesn't exist on release-0.4, backporting this won't quite work